### PR TITLE
🐛 placeholder in layout=container should not take space after hiding

### PR DESCRIPTION
**Pull request by aghassemi**
Fixes https://github.com/ampproject/amphtml/issues/18817

### CODE REVIEW

File: css/amp.css [hunk 1]
```diff
@@ -422,6 +422,16 @@ i-amphtml-sizer {
 .i-amphtml-element > [placeholder].amp-hidden {
   visibility: hidden;
 }
+/* Placeholder, by design, expands layout=container unlike other layouts where
```
Comment (cathyxz) [inline css/amp.css]: I think this means we can remove this:

https://github.com/ampproject/amphtml/blob/3b6024d439b09ee2c4fff1fd49bec6be53329a6c/css/amp.css#L503-L510

File: test/manual/amp-list-resize.amp.html [hunk 2]
```diff
@@ -79,6 +83,9 @@ <h2>AMP List Basic</h2>
     <div overflow>
       SEE MORE
     </div>
+    <div placeholder>
+      <div class="tall">PACEHOLDER</div>
```
Comment (cathyxz) [inline test/manual/amp-list-resize.amp.html]: Sorry, since you had a commit for typos already... =P
> PACEHOLDER > PLACEHOLDER

**Comment by choumx** [comment]
Thanks for looking into this. Hmm, a special case for `layout=container` is unfortunate. Do you remember why we used `visibility: hidden` instead of `display: none` in the first place?

File: css/amp.css [hunk 3]
```diff
@@ -422,6 +422,16 @@ i-amphtml-sizer {
...
+   it is position:absolute and overlays the layout box.
+   However, when it goes away it should no longer expand the parent, so instead
+   of visibility:hidden it must be display:none */
+.i-amphtml-layout-container > [placeholder].hidden,
+[layout=container] > [placeholder].hidden,
```
Comment (choumx) [inline css/amp.css]: Minor: Are the `[layout=container] > ...` rules necessary? 

I think that it targets not-built elements, but not-built elements should display the placeholder. So I think `[layout=container] > [placeholder].hidden` may not actually affect any elements in practice.

E.g. existing rules that hide the placeholder also don't use `[layout]` attribute selector:
https://github.com/ampproject/amphtml/blob/cd1d9380ce4a73250ef6f21111bbe4b8b84fdf72/css/amp.css#L421-L424

Comment (aghassemi) [reply css/amp.css]: yeah you are right. `amp-list` when it becomes container does not seem to append `.i-amphtml-layout-container`. regardless I will hold on merging this until a decision is made on https://github.com/ampproject/amphtml/issues/18849

Comment (aghassemi) [reply css/amp.css]: agreed and done.

**Comment by choumx** [comment]
@jridgewell https://github.com/ampproject/amphtml/pull/18820#issuecomment-431416793 😄 

> Do you remember why we used visibility: hidden instead of display: none in the first place?

> My guess is performance and title of 14f3c75#diff-49f8e6b8acb3722aad62bd821720af8dL131 implies that as well. visibility:hidden would not cause reflow and for size-defined elements, since placeholder is overlayed, visibility:hidden works fine.

**Pull request merged**